src/ts/misc.ts

```diff
@@ -72,13 +72,17 @@ function generateOnThisPage() {
 }
 
 (function ($) {
-    const el = document.querySelector(".header-container");
-
     const observer = new IntersectionObserver(([e]) => {
         e.target.classList.toggle("is-pinned", e.intersectionRatio < 1)
     }, { threshold: [1] });
 
-    observer.observe(el);
+    const headerContainerEl = document.querySelector(".header-container");
+    // The header-container won't be available in the registry.
+    if (!headerContainerEl) {
+        observer.observe(document.querySelector(".top-nav-bar.registry"));
```

Comment:
Do you think it'd be correct to say "if here isn't a `.header-container`, assume we're in the registry"? I only ask because I can imagine this breaking at some point. Maybe query the DOM for another class, or make the classes line up such that this pinning behavior would work everywhere? Or just try looking for both navs, and if neither is found, just bailing?

Reply:
I updated the comment slightly and also added a check for the fallback selector as well. So if neither selectors are available, the observer won't be attached to any DOM element.